packages/fxa-auth-server/fxa-oauth-server/docs/pairwise-pseudonymous-identifiers.md

```diff
@@ -0,0 +1,68 @@
+
+PPIDs allow Mozilla to enforce our privacy stance by providing each PPID enabled RP with a distinct userid for a given user. For example, a user that signs into 2 PPID enabled services would have 3 distinct userids:
+
+- FxA userid, only seen by Mozilla properties
+- PPID for service 1, known by Mozilla and service 1
+- PPID for service 2, known by Mozilla and service 2
+
+PPIDs prevents service 1 and service 2 from correlating user data with each other based on userid alone. _PPIDs only prevent correlation based on userid_, transient information such as IP address, shared 3rd party cookies, and user-agent information can still be used by RPs to correlate users. This also assumes that RPs are unable to fetch user profile information such as email address and display name that could be used as additional data points.
+
+PPIDs are returned as the `sub` claim within an [OIDC ID Token][#oidc-id-token] or [OAuth JWT Access Token][#oauth-jwt-access-token].
+
+## Further enhancing privacy through PPID rotation
+
+User privacy can be further enhanced by preventing long term profiles from being built for a given user through PPID rotation. Firefox accounts supports two methods of PPID rotation, RP initiated, and server initiated.
+
+### RP initiated PPID rotation
+
+Any PPID enabled RP that receives JWT access tokens or OIDC ID tokens can initiate PPID rotation by specifying an additional parameter, `ppid_seed`, when requesting tokens from `/token` endpoints on the OAuth server. `ppid_seed` must be an integer between 0 and 1024, and can be rotated at any time.
+
+### Server initiated PPID rotation
+
+By default, Firefox accounts does not enforce sub rotation, but for sensitive RPs where long term user profiles are undesirable, Firefox accounts can enforce periodic rotation without depending on the RP to pass in a `ppid_seed`. The default period is 6 hours.
+
+## Ensuring 3rd parties are unable to access user information
+
+An OAuth access token is a bearer token that can be presented by anyone to an OAuth service provider to access a protected resource. If a JWT access token contains a PPID `sub` claim that is meant to protect a user's privacy, but is granted the `profile` scope allows the service provider to present the same access token to the FxA profile server and learn the user's true identity. As such, tokens _MUST NOT_ be requested with the `profile` scope or any of its implicants such as `profile:uid` or `profile:email`.
+
+## PPIDs and logging
+
+PPIDs are not used by internal Mozilla properties and are not logged.
+
+## PPID generation
+
+PPIDs are generated using [HKDF][#hkdf] with the following inputs:
+
+- **km**: `client_id`.`userid_hex`.`ppid_seed||0`.`serverEnforcedRotationInput`
+- **info**: `oidc ppid sub`
+- **salt**: A secret configured by Ops
+- **len**: 8 bytes, which leads to a 16 hex character result
```

Comment:
You are right, and this was caused by me incorrectly assuming `unique.id` in configuration meant a userid. This is the bug that I pointed out that was found when I changed the tests at your requests. Instead of pulling the length from config, I now [use the length of the userid buffer](https://github.com/mozilla/fxa/blob/ed5d0305dc19343eff283a12798005e1bf7ebd3f/packages/fxa-auth-server/fxa-oauth-server/lib/jwt_sub.js#L50) that is passed in. Updating the docs to reflect that.